client/src/components/SearchOption/SearchButtons/HomeType.js

```diff
@@ -0,0 +1,123 @@
+import React, { useState } from 'react';
+import {
+  Button,
+  Checkbox,
+  FormControlLabel,
+  Menu,
+  MenuItem,
+  styled as muiStyled,
+} from '@mui/material';
+
+const StyledButton = muiStyled(Button)({
+  backgroundColor: 'white',
+  width: '130px',
+  height: '45px',
+  fontSize: '17px',
+  fontWeight: '550',
+  color: 'black',
+  '&:hover': {
+    border: '1px solid black',
+  },
+});
+
+export default function HomeType() {
+  const [anchorEl, setAnchorEl] = useState(null);
+  const [selectedTypes, setSelectedTypes] = useState([]);
+  const [buttonText, setButtonText] = useState('Home Type');
+
+  const handleClick = event => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handleClose = () => {
+    setAnchorEl(null);
+    if (selectedTypes.length === 0) {
+      setButtonText('Home Type');
+    }
+  };
+
+  const handleSelect = type => {
+    setSelectedTypes(prevSelectedTypes => {
+      if (prevSelectedTypes.includes(type)) {
+        return prevSelectedTypes.filter(t => t !== type);
+      }
+      return [...prevSelectedTypes, type];
+    });
+  };
+
+  const isSelected = type => selectedTypes.includes(type);
+
+  return (
+    <div>
+      <StyledButton onClick={handleClick}>
+        {selectedTypes.length > 0
+          ? `${selectedTypes.length} Selected`
+          : `${buttonText}`}
+      </StyledButton>
+      <Menu
+        anchorEl={anchorEl}
+        open={Boolean(anchorEl)}
+        onClose={handleClose}
+        MenuListProps={{
+          'aria-labelledby': 'home-type-button',
+        }}
+      >
+        <MenuItem>
+          <FormControlLabel
+            control={
+              <Checkbox
+                checked={isSelected('Apartment')}
+                onChange={() => handleSelect('Apartment')}
+              />
+            }
+            label="Apartment"
+          />
+        </MenuItem>
```

Comment:
It's a good idea to extract <MenuItem> ---- </MenuItem> as a component and pass the 'type' as a prop to reduce code duplication. This way, you'll only need to update the component in one place if we need to update the type enum.

Reply:
fixed